oci/oci.go

```diff
@@ -577,8 +551,13 @@ func waitContainerStop(ctx context.Context, c *Container, timeout time.Duration)
 				return
 			default:
 				// Check if the process is still around
-				err := unix.Kill(c.state.Pid, 0)
-				if err == unix.ESRCH {
+				proc, err := os.FindProcess(c.state.Pid)
+				if err != nil {
+					close(done)
+					return
+				}
+				err = proc.Signal(syscall.Signal(0))
+				if err == syscall.ESRCH {
```

Comment:
I don't think we want to do this on Windows; even if they nominally have this error, [it's a dummy value][1].  And [Window's `Signal(…)` only supports `Kill`][2], so I think we might as well use [`proc.Kill()`][3] directly.

[1]: https://github.com/golang/go/blob/b85433975aedc2be2971093b6bbb0a7dc264c8fd/src/syscall/zerrors_windows.go#L15
[2]: https://github.com/golang/go/blob/b85433975aedc2be2971093b6bbb0a7dc264c8fd/src/os/exec_windows.go#L57-L72
[3]: https://godoc.org/os#Process.Kill

Reply:
can we get to this once we have tests?

Reply:
Ah, this is the wrong check for `proc.Signal` on Unix, which [internally translates][1] to [a private `errFinished][2].  I'm not sure what the best way to fix that is (compare `err.Error()`?), but I bet that's what's going on.

[1]: https://github.com/golang/go/blob/010579c2377654e575d46857ec7dc77bab586438/src/os/exec_unix.go#L72-L73
[2]: https://github.com/golang/go/blob/010579c2377654e575d46857ec7dc77bab586438/src/os/exec_unix.go#L53